libraries/ms-common/src/bootloader_can.c

```diff
@@ -0,0 +1,51 @@
+#include "bootloader_can.h"
+
+#include <stdint.h>
+#include <string.h>
+
+#include "can.h"
+#include "can_msg_defs.h"
+#include "can_transmit.h"
+#include "status.h"
+
+#define CLIENT_SCRIPT_CONTROLLER_BOARD_ID 0
+
+static BootloaderCallback s_callback = NULL;
+static void *s_context = NULL;
+
+StatusCode bootloader_can_init(CanStorage *storage, CanSettings *settings) {
+  settings->device_id = SYSTEM_CAN_DEVICE_BOOTLOADER;
+
+  return can_init(storage, settings);
+}
+
+StatusCode bootloader_can_transmit(uint16_t board_id, uint8_t *data, size_t len,
+                                   bool is_start_message) {
+  CanMessage message = { .source_id = SYSTEM_CAN_DEVICE_BOOTLOADER,
+                         .msg_id = board_id,
+                         .type = CAN_MSG_TYPE_DATA,
+                         .dlc = len };
+  // copy the message data over to the message to be transmitted
+  memcpy(message.data_u8, data, 8);

```

Comment:
We should only copy `len` bytes instead of 8 bytes - otherwise we'll segfault if you pass in a `data` array of <8 bytes. (Also, mention in the docs in the header that `len` has to be <=8, and maybe return `STATUS_CODE_INVALID_ARGS` early if `len > 8`.)